src/ethernet/ip_device.cpp

```diff
@@ -0,0 +1,310 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include "ip_device.hh"
+#include "api.h"
+#include <librealsense2-net/rs_net.h>
+
+#include <ipDeviceCommon/Statistic.h>
+#include <list>
+
+#include <chrono>
+#include <thread>
+
+std::string sensors_str[] = {"depth", "color"};
+
+//WA for stop
+void ip_device::recover_rtsp_client(int sensor_index)
+{
+    remote_sensors[sensor_index]->rtsp_client = RsRTSPClient::getRtspClient(std::string("rtsp://" + ip_address + ":8554/" + sensors_str[sensor_index]).c_str(), "ethernet_device");
+
+    ((RsRTSPClient *)remote_sensors[sensor_index]->rtsp_client)->initFunc(&rs_rtp_stream::get_memory_pool());
+    ((RsRTSPClient *)remote_sensors[sensor_index]->rtsp_client)->getStreams();
+}
+
+ip_device::~ip_device()
+{
+    is_device_alive = false;
+
+    for (int remote_sensor_index = 0; remote_sensor_index < NUM_OF_SENSORS; remote_sensor_index++)
+    {
+        stop_sensor_streams(remote_sensor_index);
+    }
+
+    if (sw_device_status_check.joinable())
+    {
+        sw_device_status_check.join();
+    }
+    std::cout << "destroy ip_device\n";
+}
+
+void ip_device::stop_sensor_streams(int sensor_index)
+{
+    for (long long int key : remote_sensors[sensor_index]->active_streams_keys)
+    {
+        std::cout << "\t@@@ stopping stream [uid:key] " << streams_collection[key].get()->m_rs_stream.uid << ":" << key << "]" << std::endl;
+        streams_collection[key].get()->is_enabled = false;
+        if (inject_frames_thread[key].joinable())
+            inject_frames_thread[key].join();
+    }
+    remote_sensors[sensor_index]->active_streams_keys.clear();
+}
+
+ip_device::ip_device(std::string ip_address, rs2::software_device sw_device)
+{
+    this->ip_address = ip_address;
+    this->is_device_alive = true;
+
+    //init device data
+    init_device_data(sw_device);
+}
+
+std::vector<rs2_video_stream> ip_device::query_streams(int sensor_id)
+{
+    std::cout << "Querry Sensors\n";
+    std::vector<rs2_video_stream> streams;
+
+    if (remote_sensors[sensor_id]->rtsp_client == NULL)
```

Comment:
Are these two checks different?

Reply:
@dorodnic , not sure I got your comment here.